util/types/time.go

```diff
@@ -982,6 +982,7 @@ func parseDateTimeFromNum(num int64) (Time, error) {
 
 	// Check datetime type.
 	if num >= 10000101000000 {
+		t.Type = mysql.TypeDatetime
```

Comment:
The default type of `ZeroDate` in line 977 is `mysql.TypeDate`,
but for num >= 10000101000000 which contains a time part,
the type should be changed to mysql.TypeDatetime. @hanfei1991